### PR TITLE
⚡️ Ignore case sensitive for `flutter_style_todos`

### LINKED ISSUE #3616: `flutter_style_todos` not covered all TODOs
**Describe the issue**

We want to force all recognizable TODOs to follow the `flutter_style_todos` in our projects, but only uppercase `TODO` can be applied, they are all recognized by the editor.

image

```dart
// TODO something <- Worked
// Todo something <- Not working
// todo something <- Not working
```

It could be checked with a lowercase string.

### PR DESCRIPTION
**Pull request by AlexV525**
Fixes #3616

### CODE REVIEW

**Comment by pq** [comment]
As per @goderbauer's comment in https://github.com/dart-lang/linter/issues/3616#issuecomment-1230548924, I'm thinking we want to treat this lint as WAI and close this out.

Feel free to chime in on the issue and if you can change the Flutter team's mind, we can re-open.

Thanks so much for the contribution!

**Comment by AlexV525** [comment]
@pq Can we re-open this since we have some more detailed statements discussed in the issue?

**Comment by pq** [comment]
While we're here, we might consider re-writing this check to not use regular expressions.  It's pretty slow as is and adding case insensitivity to the regexp slows it down even more...

Before this change:

image

After:

image

(Note that absolute timings aren't reliable but I find the relative ranking useful.)

**Comment by pq** [comment]
You'll want to update tests too.

```

00:55 +226: rule dart flutter_style_todos

// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// test w/ `dart test -N flutter_style_todos`
// TODO something //LINT
// Todo something //LINT
// todo something //LINT
CompilationUnitImpl [] // Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file


/home/runner/work/linter/linter/test_data/rules/flutter_style_todos.dart 7:1 [lint] Use Flutter TODO format: // TODO(username): message, https://URL-to-issue.
// TODO something //LINT
^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/work/linter/linter/test_data/rules/flutter_style_todos.dart 8:1 [lint] Use Flutter TODO format: // TODO(username): message, https://URL-to-issue.
// Todo something //LINT
^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/work/linter/linter/test_data/rules/flutter_style_todos.dart 9:1 [lint] Use Flutter TODO format: // TODO(username): message, https://URL-to-issue.
// todo something //LINT
^^^^^^^^^^^^^^^^^^^^^^^^

null files analyzed, 3 issues found, in null ms.

00:55 +226 -1: rule dart flutter_style_todos [E]

  Expected: matches [] unordered
    Actual: [
              Annotation:[LINT]: "Use Flutter TODO format: // TODO(username): message, https://URL-to-issue." (line: 7) - [1:24],
              Annotation:[LINT]: "Use Flutter TODO format: // TODO(username): message, https://URL-to-issue." (line: 8) - [1:24],
              Annotation:[LINT]: "Use Flutter TODO format: // TODO(username): message, https://URL-to-issue." (line: 9) - [1:24]
            ]
     Which: has too many elements (3 > 0)
```

https://github.com/dart-lang/linter/runs/8096820336?check_suite_focus=true

**Comment by AlexV525** [comment]
> You'll want to update tests too.

I was checking this. The rule test doesn't seem to support `// LINT` after TODOs, as they'll be recognized as a part of the TODO. Any thoughts? @pq

File: test/rules/flutter_style_todos_test.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
```
Comment (pq) [inline test/rules/flutter_style_todos_test.dart:1]: You'll want to add this to `all.dart`.

File: test_data/rules/flutter_style_todos.dart [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
```
Comment (pq) [inline test_data/rules/flutter_style_todos.dart:1]: I'd remove this test data file and just test in the reflective test style.

**Review by bwilkerson — APPROVED** [review]

**Comment by pq** [comment]
Hey @AlexV525!  This is great, thanks.  I'm going to hold on landing this until after 1.28.0.  That release is blocking an analyzer roll and I don't want to jeopardize that with any other changes that could cause issues with our internal roll.

That said, we can get working on another release soonish after.

Cheers!

File: test/rules/flutter_style_todos_test.dart [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+// Todo something
+// todo something
+// TODO(somebody) something
+// TODO: something
+// Todo(somebody): something
+// todo(somebody): something
+// ToDo(somebody): something
+// TODO(somebody): something
+''',
+      [
```
Comment (pq) [inline test/rules/flutter_style_todos_test.dart:33]: I can't actually tell from looking at this list where the expected failures are.  Could you consider (at least) breaking these up into one test for OK patterns and another for bad ones?

**Comment by pq** [comment]
Not necessarily for this PR but I still think it'd be great to look at improving the performance of this lint. 🤔

**Review by pq — CHANGES_REQUESTED** [review]
Assuming I understand where the failures are supposed to be (see comment) LGTM! :D

**Comment by AlexV525** [comment]
> Not necessarily for this PR but I still think it'd be great to look at improving the performance of this lint. 🤔

We can maybe keep it tracked in separate issues. I will file a new one after it's landed.

**Review by pq — APPROVED** [review]

**Pull request merged**